pkg/controller/deployment/deployment_controller.go

```diff
@@ -959,6 +957,10 @@ func (dc *DeploymentController) calculateStatus(allRCs []*api.ReplicationControl
 }
 
 func (dc *DeploymentController) scaleRCAndRecordEvent(rc *api.ReplicationController, newScale int, deployment extensions.Deployment) (*api.ReplicationController, error) {
+	// No need to scale
+	if rc.Spec.Replicas == newScale {
+		return rc, nil
```

Comment:
Some callers (eg. line 805, above) want to know if scaling occurred. They would determine that by comparing replicas from before and after?